src/payment/strategies/klarna/klarna-payment-strategy.ts

```diff
@@ -65,9 +73,9 @@ export default class KlarnaPaymentStrategy implements PaymentStrategy {
 
         const { payment: { paymentData, ...paymentPayload } } = payload;
 
-        return this._authorize()
+        return this._authorize(paymentPayload.methodId)
             .then(({ authorization_token: authorizationToken }) => this._store.dispatch(
-                this._remoteCheckoutActionCreator.initializePayment(paymentPayload.methodId, { authorizationToken })
+                this._remoteCheckoutActionCreator.initializePayment('klarna', { authorizationToken })
```

Comment:
isnt `paymentPayload.methodId === 'klarna' ` ? do we need to pass it hard-coded?

Reply:
They are hard-coding this because Klarna V2 is a multi-option gateway, meaning method ID will be something like `pay_later`.